### DISCUSSION TITLE
Voice to camera speaker

### DISCUSSION THREAD

**Discussion by slovenec88** [tsightler/ring-mqtt#591] [Feature Requests]
Would it be possible to send a pre-recorded sound or TTS directly to a camera speaker? It would be cool to have some announcements on the outside for example when gates are opening/closing or garage doors are about to automatically close?

**Comment by tsightler** [comment]
The only way to send audio, at least that I'm aware of, is to create a live stream and enable two-way audio and feed the audio back to the return side.  The API supports it, and there's [some example code](https://github.com/dgreif/ring/blob/main/packages/examples/return-audio-example.ts).

However, there's lots of stuff that would have to be figured out, like how a user would configure what files could be played back (maybe a folder where you can drop the files that is automatically scanned during startup), then you'd need some MQTT topic to trigger the playback of a specific file, etc.  It's a neat idea, but probably not something I'll be adding anytime soon.  Maybe if I get super bored one day.
